geomaster/scripts/render_mesh.py

```diff
@@ -0,0 +1,218 @@
+            camera_space_normal = camera_space_normal / normal_norm
+            camera_space_normal[background_mask.squeeze(-1)] = 0
+            camera_space_normal = -1*camera_space_normal
+            save_image(camera_space_normal, save_path, i+1)
+            save_camera_matrix(mv[i], save_path, i+1)
+
+        # Clear CUDA cache after processing
+        torch.cuda.empty_cache()
+
+        # Log completion
+        log_processed_id(log_file, obj_id)
+        print(f"Processed {obj_id} on GPU {gpu_id} (Process {process_idx})")
+        
+    except Exception as e:
+        print(f"Error processing {ply_path} on GPU {gpu_id} (Process {process_idx}): {str(e)}")
+        # Clear CUDA cache in case of error
+        torch.cuda.empty_cache()
+
+def worker(task_queue, gpu_id, output_dir, log_file, process_idx):
+    """Worker process function."""
+    try:
+        while True:
+            try:
+                ply_path = task_queue.get_nowait()
+                if ply_path is None:
+                    break
+                process_ply_file(ply_path, gpu_id, output_dir, log_file, process_idx)
+            except queue.Empty:
+                break
+    except Exception as e:
+        print(f"Worker error on GPU {gpu_id} (Process {process_idx}): {str(e)}")
+    finally:
+        # Ensure CUDA cache is cleared when worker exits
+        torch.cuda.empty_cache()
+
+def main(args):
+    """Main function with multi-process support on single GPU."""
+    # Verify CUDA is available
+    if not torch.cuda.is_available():
+        raise RuntimeError("No CUDA devices available")
+    
+    # Initialize multiprocessing with spawn method for CUDA compatibility
+    tmp.set_start_method('spawn', force=True)
+    
+    # Create task queue
+    task_queue = mp.Queue()
+    
+    # Collect all PLY files
+    ply_files = []
+    for root, dirs, files in os.walk(args.input_dir):
+        for file in files:
+            if file.endswith('.ply'):
+                ply_path = os.path.join(root, file)
+                ply_files.append(ply_path)
+    
+    # Put all tasks in the queue
+    for ply_path in ply_files:
+        task_queue.put(ply_path)
+    
+    # Add termination signals
+    for _ in range(PROCESSES_PER_GPU):
+        task_queue.put(None)
+    
+    # Create and start worker processes
+    processes = []
+    gpu_id = 0  # Using single GPU (cuda:0)
+    
+    for process_idx in range(PROCESSES_PER_GPU):
+        p = mp.Process(
+            target=worker,
+            args=(task_queue, gpu_id, args.output_dir, args.log_file, process_idx)
+        )
+        p.start()
+        processes.append(p)
+    
+    # Wait for all processes to complete
+    for p in processes:
+        p.join()
+    
+    print("All processing complete!")
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description="Batch render .ply files with multi-processing on single GPU.")
+    parser.add_argument('--input_dir', '-i', type=str, required=True, 
```

Comment:
For render_mesh.py, the input should be a single mesh. We have better implement two functions, one is render_mesh.py, another is render_meshes.py.